sdk/monitor/Azure.Monitor.Ingestion/tests/MonitorIngestionLiveTest.cs

```diff
@@ -148,16 +149,19 @@ public async Task ValidInputFromArrayAsJsonWithMultiBatchWithGzip()
         }
 
         [AsyncOnly]
+        [LiveOnly]
         [Test]
         public async Task ConcurrencyMultiThread()
         {
-            var policy = new ConcurrencyCounterPolicy(10);
+            var policy = new ConcurrencyCounterPolicy(8);
             LogsIngestionClient client = CreateClient(policy);
-
+            LogsIngestionClient.Compression = "gzip";
+            LogsIngestionClient.SingleUploadThreshold = Kb;
             // Make the request
             LogsUploadOptions options = new LogsUploadOptions();
-            options.MaxConcurrency = 10;
-            Response response = await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, GenerateEntries(8, Recording.Now.DateTime), options).ConfigureAwait(false);
+            options.MaxConcurrency = 8;
+            Response response = await client.UploadAsync(TestEnvironment.DCRImmutableId, TestEnvironment.StreamName, GenerateEntries(8000, Recording.Now.DateTime), options).ConfigureAwait(false);
+            Assert.Greater(policy.MaxCount, 1);
```

Comment:
Does this fail if you move the line back into the foreach loop?

Reply:
It fails! Concurrency is always 1